### PR TITLE
Add Option::is_none_or

### PR DESCRIPTION
**Pull request by RalfJung**
ACP: https://github.com/rust-lang/libs-team/issues/212

### CODE REVIEW

**Comment by RalfJung** [comment]
@Veykril  what's the best way to deal with this warning in RA?

**Comment by Veykril** [comment]
Adding a `#[allow(unstable_name_collisions)]` on the usage should probably be fine

**Comment by RalfJung** [comment]
There's probably a bunch of usage so I guess that has to be on the crate level.

**Comment by Veykril** [comment]
No that's the only one

File: compiler/rustc_const_eval/src/interpret/eval_context.rs [hunk 1]
```diff
@@ -1057,7 +1057,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
 
                 ty::Str | ty::Slice(_) | ty::Dynamic(_, _, ty::Dyn) | ty::Foreign(..) => false,
 
-                ty::Tuple(tys) => tys.last().iter().all(|ty| is_very_trivially_sized(**ty)),
+                ty::Tuple(tys) => tys.last().is_none_or(|ty| is_very_trivially_sized(*ty)),
```
Comment (workingjubilee) [inline compiler/rustc_const_eval/src/interpret/eval_context.rs:1060]: was like "wait but this used to iterate" for a second and then I realized it was iterating an Option...

File: src/tools/miri/src/shims/foreign_items.rs [hunk 2]
```diff
@@ -396,12 +396,12 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
...
                     // allocation, and bigger than the previously promised alignment, then set it.
                     if align > alloc_align
-                        && !this
+                        && this
                             .machine
                             .symbolic_alignment
                             .get_mut()
                             .get(&alloc_id)
-                            .is_some_and(|&(_, old_align)| align <= old_align)
+                            .is_none_or(|&(_, old_align)| align > old_align)
```
Comment (workingjubilee) [inline src/tools/miri/src/shims/foreign_items.rs:404]: ...wow this sure is... yeah I see why you bothered to make the PR.

**Review by workingjubilee — COMMENTED** [review]
significant readability improvement in some cases! lgtm.

**Comment by CryZe** [comment]
Is it normal for it to be merged without a tracking issue?

**Comment by RalfJung** [comment]
Eh... no.^^
https://github.com/rust-lang/rust/pull/126384

**Pull request merged**